src/printer-markdown.js

```diff
@@ -88,7 +88,7 @@ function genericPrint(path, options, print) {
       const nextNode = parentNode.children[index + 1];
 
       // special prefix that may cause different meaning
-      if (nextNode && nextNode.value.match(/^(#{1,6}|>+)|^(-|\+|\*)$/)) {
+      if (nextNode && /^(#{1,6}|>)|^[-+*]$/.test(nextNode.value)) {
```

Comment:
Actually, I just realized that the `{1,6}` check for `#` is unnecessary, too: `/^[#>]|^[-+*]$/`

Or did you mean to only match up to 6 `#`?

Now that I'm thinking about it, is there a reason that we check for _prefix_ for `#` and `>` but _exact string_ for `-`, `+` and `*`? Just curios (and a tiny bit confused by the comment the line above).

Reply:
> Or did you mean to only match up to 6 #?

Yes, `#######` won't be considered heading.

---

Just realized we can also check the exact string for headings since [CommonMark](http://spec.commonmark.org/0.25/#atx-headings) does not allow the space after `#`.

special prefix

```md
- 123
* 123
+ 123
>123
# 123
```

not special prefix

```md
-- 123
** 123
++ 123
#123
```

Uh, would be better to call it leading char?